cfme/infrastructure/provider/openstack_infra.py

```diff
@@ -106,3 +119,27 @@ def node_exist(self, name='my_node'):
         node_uuid = str(nodes_dict[name])
         for db_node in query.all():
             return db_node.hosts.name == str(node_uuid.uuid)
+
+
+@navigator.register(OpenstackInfraProvider, 'All')
+class CloudInfraProviderAll(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+
+    def step(self):
+        self.prerequisite_view.navigation.select('Compute', 'Infrastructure', 'Providers')
+
+
+@navigator.register(OpenstackInfraProvider, 'Details')
+class CloudInfraProviderDetails(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        sel.click(Quadicon(self.obj.name, self.obj.quad_name))
+
+
+@navigator.register(OpenstackInfraProvider, 'ProviderNodes')
+class ProviderNodes(CFMENavigateStep):
```

Comment:
@izapolsk nodes are unique feature of openstack, it might appear as hosts in other providers.